src/analysis/allman.d

```diff
@@ -0,0 +1,221 @@
+// Distributed under the Boost Software License, Version 1.0.
+//	  (See accompanying file LICENSE_1_0.txt or copy at
+//			http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.allman;
+
+import dparse.lexer;
+import dparse.ast;
+import analysis.base : BaseAnalyzer;
+import dsymbol.scope_ : Scope;
+
+/**
+Checks for the allman style (braces should be on their own line)
+
+------------
+if (param < 0) {
+
+}
+------------
+
+should be
+
+------------
+if (param < 0)
+{
+
+}
+------------
+*/
+class AllManCheck : BaseAnalyzer
+{
+	///
+	this(string fileName, const(ubyte)[] code, bool skipTests = false)
+	{
+		super(fileName, null, skipTests);
+		this.code = code;
+	}
+
+	override void visit(const WhileStatement st)
+	{
+		if (st.declarationOrStatement !is null)
+			checkForBrace(st.declarationOrStatement, st.expression.line, st.expression.column);
+	}
+
+	override void visit(const ForeachStatement st)
+	{
+		checkForBrace(st.declarationOrStatement, st.low.line, st.low.column);
+	}
+
+	override void visit(const ForStatement st)
+	{
+		checkForBrace(st.declarationOrStatement, st.test.line, st.test.column);
+	}
+
+	override void visit(const DoStatement st)
+	{
+		// the DoStatement only knows about the line and column of the expression
+		checkForBrace(st.statementNoCaseNoDefault, 0, 0);
```

Comment:
For 

```d
do {

} while(true)
```

the AST is:

```
<doStatement>
  <expression>
    <unaryExpression>
      <primaryExpression>
        <true/>
      </primaryExpression>
    </unaryExpression>
  </expression>
  <statementNoCaseNoDefault>
    <blockStatement>
      <declarationsAndStatements>
        <declarationOrStatement>
          <statement>
            <statementNoCaseNoDefault>
              <expressionStatement>
                <expression>
                  <unaryExpression>
                    <primaryExpression>
                      <identifierOrTemplateInstance>
                        <identifier>i</identifier>
                      </identifierOrTemplateInstance>
                    </primaryExpression>
                    <suffix>++</suffix>
                  </unaryExpression>
                </expression>
              </expressionStatement>
            </statementNoCaseNoDefault>
          </statement>
        </declarationOrStatement>
      </declarationsAndStatements>
    </blockStatement>
  </statementNoCaseNoDefault>
</doStatement>
```

and none of the elements contains information about the line and column:

DoStatement: https://github.com/Hackerpilot/libdparse/blob/master/src/dparse/ast.d#L1467
BlockStatement: https://github.com/Hackerpilot/libdparse/blob/master/src/dparse/ast.d#L959
StatementNoCaseNoDefault: https://github.com/Hackerpilot/libdparse/blob/master/src/dparse/ast.d#L2249